src/v/cluster/topics_frontend.cc

```diff
@@ -457,6 +460,46 @@ ss::future<topic_result> topics_frontend::do_create_topic(
               ->remote_partition_count;
     }
 
+    if (assignable_config.is_recovery_enabled()) {
```

Comment:
nit: maybe add an assert that `is_read_replica` and `is_recovery_enabled` are not true at the same time? Cause otherwise `remote_topic_properties` from recovery will overwrite `remote_topic_properties` from read replica

Reply:
fixed this in a different way by validating the flags in the kafka layer